sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1307,6 +1308,69 @@ class AdaptiveQueryExecSuite
     }
   }
 
+  test("SPARK-35455: Enhance EliminateUnnecessaryJoin - single join") {
+    withSQLConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true",
+      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1") {
+      Seq(
+        // left semi join and empty left side
```

Comment:
We can't optimize this before this PR?

Reply:
yeah, we cann't. Before we only check right side with `LeftSemi/LeftAnti`.

And the test should use different column to do filter and join  in case of `InferFiltersFromConstraints` make right side empty.  Updated it.